### ISSUE TITLE
GPUImageLookupFilter not working on first try

### ISSUE THREAD

**Issue by danlozano** [BradLarson/GPUImage#2142]
I've  been trying to fix this issue for weeks now to no success.

I have a photo editing extension I'm working on for Mac & iOS. It has several filters like AlphaBlend, Overlay, etc... They all work perfectly.

But for some reason the first time I open my app, the GPUImageLookupFilter "works" as in it DOES return an image, but the image is the same image as the input. If I close the app and open it again, it starts working perfectly.

It almost always works, but whenever I restart my computer or open the app again, the GPUImageLookupFilter does NOT work on the first try.

I've debugged it, and it is correctly loading my LUT images, the GPUImageLookupFilter IS returning an image, everything seems to be working, it just happens to be that the image from "imageFromCurrentFrameBuffer" it returns the same image (without the LUT being applied), on first try.

I'm correctly building the filter chain, calling "useNextFrameForImageCapture" on the last filter, calling proccesImage on all inputs, and calling "imageFromCurrentFrameBuffer" on the last filter.

I guess it has something to do with the first time I try to do this maybe some part of the framework doesn't get a chance to load correctly? It is very weird.

**Comment by danlozano** [comment]
Have had no luck fixing this. 

I thought for a moment that the problem was that processImage was taking too long, so imageFromCurrentFrameBuffer returned the same image. Tried using the async processImageWithCompletionHandler method, but the problem persists.

I did figure something out though. On first try, the lookup filter does not work. On subsequent tries, it does work. I realized that if I do a "sudo purge" in the terminal, which clears up ram memory, the lookup filter does not work again.

So something to do with the original image or LUT image only working on 2nd + attempts after it has been loaded up for the first time.

Would appreciate any help, I must send update to app store. @BradLarson thanks for the awesome framework!

**Comment by danlozano** [comment]
Fixed the issue, just in case anyone else has the same problem. I am using Swift and this is a Mac OS X app.

The GPUImageLookupFilter I was using was being stored in an instance variable, like so:

lazy var lookupFilter = GPUImageLookupFilter()

I added the lazy part to only have to instantiate it whenever it's used, and not before. But apparently this is what caused the bug. On first use it did not work. On consequent launches it did. I don't know if this is the expected behavior or this is some kind of bug. 

Removed the lazy part and now it's working. I only have this issue with the lookup filter, all other filters are lazy and always work.

@BradLarson
